src/client_side.cc

```diff
@@ -3422,10 +3416,7 @@ clientStartListeningOn(AnyP::PortCfgPointer &port, const RefCount< CommCbFunPtrC
     // Fill out a Comm::Connection which IPC will open as a listener for us
     port->listenConn = new Comm::Connection;
     port->listenConn->local = port->s;
-    port->listenConn->flags =
-        COMM_NONBLOCKING |
-        (port->flags.tproxyIntercept ? COMM_TRANSPARENT : 0) |
-        (port->flags.natIntercept ? COMM_INTERCEPTION : 0);
```

Comment:
Lets pull this hidden change to the Comm API out to a separate PR as well. There are some careful TPROXY and NAT tests that need to be done, so I will take this on today.